test/edu/csus/ecs/pc2/core/execute/ExecutableTest.java

```diff
@@ -1198,6 +1198,42 @@ public void testSubstituteFiles() throws Exception {
         
         assertEquals("Expected substitution", actual, "foo ISumit.java");
     }
+    
+    /**
+     * test {:ifsuff=xxx} substitution.
+     * 
+     * @throws Exception
+     */
+    public void testCondSuffixSubstitute() throws Exception {
+        
+        String executeDirectoryName = getOutputDataDirectory(getName());
+        ensureDirectory(executeDirectoryName);
+        
+        String sumitFilename = getSamplesSourceFilename("ISumit.java");
+        ClientId submitter = contest.getAccounts(Type.TEAM).lastElement().getClientId();
+        Problem problem = createHelloProblemNoJudgesData(contest);
+        
+        Run run = createRun(submitter, javaLanguage, problem, 2, 2);
+        assertFileExists(sumitFilename);
+        
+        RunFiles runFiles = new RunFiles(run, sumitFilename);
+        
+        Executable executable = new Executable(contest, controller, run, runFiles);
+        executable.setExecuteDirectoryName(executeDirectoryName);
+        executable.setUsingGUI(false);
+        
+        String actual = executable.substituteAllStrings(run, "foo {:basename}{:ifsuffix=it}");
+        
+        assertEquals("Expected substitution", actual, "foo ISumit");
```

Comment:
these assertEquals parameters are backwards... they are ordered "Expected" actualStr, expectedStr
All the other assertEquals are ordered "Expected ", expectedStr, actualStr